critiquebrainz/data/dump_manager.py

```diff
@@ -221,18 +217,51 @@ def public(location, rotate=False):
         f.write(str(db.SCHEMA_VERSION))
 
     # BASE ARCHIVE
-    # Archiving stuff that is independent from licenses (users, licenses)
+    # Contains all license independent data (licenses, users)
+    base_archive_path = create_base_archive(dump_dir, temp_dir)
+    print(base_archive_path)
+
+    # 1. COMBINED
+    # Archiving all reviews (any license)
+    review_dump_path = create_reviews_archive(dump_dir, temp_dir)
+    print(review_dump_path)
+
+    # 2. SEPARATE
+    # Creating separate archives for each license
+    for license in db_license.list_licenses():
+        review_dump_path = create_reviews_archive(dump_dir, temp_dir, license_id=license['id'])
+        print(review_dump_path)
+
+    shutil.rmtree(temp_dir)  # Cleanup
+
+    if rotate:
+        print("Removing old dumps (except two latest)...")
+        remove_old_archives(location, "[0-9]+-[0-9]+", is_dir=True)
+
+    print("Done!")
+
+
+def create_base_archive(dump_dir, temp_dir):
+    """Creates a dump of all license-independent information: (users, license)."""
+
     with tarfile.open(os.path.join(dump_dir, "cbdump.tar.bz2"), "w:bz2") as tar:
         base_archive_dir = os.path.join(temp_dir, 'cbdump')
         create_path(base_archive_dir)
 
         # Dumping tables
         base_archive_tables_dir = os.path.join(base_archive_dir, 'cbdump')
         create_path(base_archive_tables_dir)
-        with open(os.path.join(base_archive_tables_dir, 'user_sanitised'), 'w') as f:
-            cursor.copy_to(f, '"user"', columns=('id', 'created', 'display_name', 'musicbrainz_id'))
-        with open(os.path.join(base_archive_tables_dir, 'license'), 'w') as f:
-            cursor.copy_to(f, 'license', columns=_TABLES["license"])
+        with db.engine.connect() as connection:
+            with connection.begin() as transaction:
+                cursor = connection.connection.cursor()
+                try:
+                    with open(os.path.join(base_archive_tables_dir, 'user_sanitised'), 'w') as f:
+                        cursor.copy_to(f, '"user"', columns=('id', 'created', 'display_name', 'musicbrainz_id'))
+                    with open(os.path.join(base_archive_tables_dir, 'license'), 'w') as f:
+                        cursor.copy_to(f, 'license', columns=_TABLES["license"])
+                except Exception:
+                    print('Error while copying tables')
```

Comment:
You should log the error as well. Also, the message should be something like `error while copying tables during creation of base archive` so that it is easier to understand where the error took place.